File: 7-kyu/bits-battle/python/solution.py
def bits_battle(numbers):
    odds = [n for n in numbers if n % 2 == 1]
    evens = [n for n in numbers if n % 2 == 0 and n != 0]
    odd_ones_count = sum(f'{n:b}'.count('1') for n in odds)
    even_zeroes_count = sum(f'{n:b}'.count('0') for n in evens)
    return 'odds win' if odd_ones_count > even_zeroes_count \
        else 'evens win' if even_zeroes_count > odd_ones_count else 'tie'

File: 7-kyu/bits-battle/javascript/solution.js
const bitsBattle = numbers => {
  const odds = numbers.filter(number => number % 2 === 1).map(number => number.toString(2));
  const evens = numbers.filter(number => number % 2 === 0 && number !== 0).map(number => number.toString(2));
  const oddsOneCounts = odds.map(number => number.split("").filter(digit => digit === "1").length).reduce((acc, nxt) => acc + nxt, 0);
  const evensZeroCounts = evens.map(number => number.split("").filter(digit => digit === "0").length).reduce((acc, nxt) => acc + nxt, 0);
  return oddsOneCounts > evensZeroCounts ? "odds win" : oddsOneCounts === evensZeroCounts ? "tie" : "evens win";
};
